pybamm/models/submodels/transport_efficiency/general_transport_efficiency.py

```diff
@@ -0,0 +1,83 @@
+#
+# Class for Bruggemantransport_efficiency
+#
+import pybamm
+from .base_transport_efficiency import BaseModel
+
+
+class GeneralTransportEfficiency(BaseModel):
+    """Submodel for transport_efficiency
+
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+    component : str
+        The material for the model ('electrolyte' or 'electrode').
+    options : dict, optional
+        A dictionary of options to be passed to the model.
+    """
+
+    def __init__(self, param, component, options=None):
+        super().__init__(param, component, options=options)
+
+    def _tortuosity_factor_model(self, eps_k):
```

Comment:
can Bruggeman and tortuosity factor go in this method too, or are they treated differently for some reason?

Reply:
I don't think it reduces total lines of code as you would have to do the if statement on the domain inside the sub-function

Reply:
They are using additional specific parameters whereas the other ones just use porosity

Reply:
Makes sense, got it